test/e2e/testsuites/dynamically_provisioned_azuredisk_tag_tester.go

```diff
@@ -0,0 +1,101 @@
+	"sigs.k8s.io/azuredisk-csi-driver/test/utils/azure"
+	"sigs.k8s.io/azuredisk-csi-driver/test/utils/credentials"
+)
+
+// DynamicallyProvisionedAzureDiskWithTag will provision required StorageClass(es), PVC(s) and Pod(s)
+// Waiting for the PV provisioner to create azuredisk
+// Testing if azuredisk contains tag
+type DynamicallyProvisionedAzureDiskWithTag struct {
+	CSIDriver              driver.DynamicPVTestDriver
+	Pods                   []PodDetails
+	StorageClassParameters map[string]string
+	Tags                   string
+}
+
+func (t *DynamicallyProvisionedAzureDiskWithTag) Run(client clientset.Interface, namespace *v1.Namespace) {
+	t.StorageClassParameters["tags"] = t.Tags
+	for _, pod := range t.Pods {
+		tpod, cleanup := pod.SetupWithDynamicVolumes(client, namespace, t.CSIDriver, t.StorageClassParameters)
+		//defer must be called here for resources not get removed before using them
+		for i := range cleanup {
+			defer cleanup[i]()
+		}
+		ginkgo.By("deploying the pod")
+		tpod.Create()
+		defer tpod.Cleanup()
+
+		ginkgo.By("checking that the pod is running")
+		tpod.WaitForRunning()
+
+		ginkgo.By("checking whether azuredisk contains tag")
+
+		//Get diskURI from pv information
+		pvcname := tpod.pod.Spec.Volumes[0].VolumeSource.PersistentVolumeClaim.ClaimName
+		pvc, err := client.CoreV1().PersistentVolumeClaims(namespace.Name).Get(context.Background(), pvcname, metav1.GetOptions{})
+		framework.ExpectNoError(err, fmt.Sprintf("Error getting pvc for azuredisk %v", err))
+
+		pvname := pvc.Spec.VolumeName
+		pv, _ := client.CoreV1().PersistentVolumes().Get(context.Background(), pvname, metav1.GetOptions{})
+		diskURI := pv.Spec.PersistentVolumeSource.CSI.VolumeHandle
+		diskName, err := azuredisk.GetDiskName(diskURI)
+		framework.ExpectNoError(err, fmt.Sprintf("Error getting diskName for azuredisk %v", err))
+		resourceGroup, err := azuredisk.GetResourceGroupFromURI(diskURI)
+		framework.ExpectNoError(err, fmt.Sprintf("Error getting resourceGroup for azuredisk %v", err))
+
+		creds, err := credentials.CreateAzureCredentialFile(false)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		azureClient, err := azure.GetAzureClient(creds.Cloud, creds.SubscriptionID, creds.AADClientID, creds.TenantID, creds.AADClientSecret)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		//get disk information
+		disksClient, err := azureClient.GetAzureDisksClient()
+		framework.ExpectNoError(err, fmt.Sprintf("Error getting client for azuredisk %v", err))
+		disktest, err := disksClient.Get(context.Background(), resourceGroup, diskName)
+		framework.ExpectNoError(err, fmt.Sprintf("Error getting disk for azuredisk %v", err))
+		test, err := util.ConvertTagsToMap(t.Tags)
+		framework.ExpectNoError(err, fmt.Sprintf("Error getting tag %v", err))
+		test["created-by"] = "kubernetes-azure-dd"
+
+		for k, v := range test {
+			_, ok := disktest.Tags[k]
+			if !ok || *disktest.Tags[k] != v {
+				err := fmt.Errorf("wrong")
```

Comment:
what about use `framework.ExpectEqual(*disktest.Tags[k], v)`?

Reply:
If disktest.Tags don't contain k,there will be a panic using *disktest.Tags[k], maybe I could fix it as this ,I think it is the same as before.
```
for k, v := range test {
			_, ok := disktest.Tags[k]
			if !ok || framework.ExpectEqual(*disktest.Tags[k], v) {
				err := fmt.Errorf("wrong")
				framework.ExpectNoError(err, fmt.Sprintf("tag not equal: %v", err))
			}
		}
```